server/testutil/testserver/testserver.go

```diff
@@ -0,0 +1,116 @@
+package testserver
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"sync"
+	"testing"
+	"time"
+
+	bazelgo "github.com/bazelbuild/rules_go/go/tools/bazel"
+)
+
+const (
+	// readyCheckPollInterval determines how often to poll BuildBuddy server to check
+	// whether it's up and running.
+	readyCheckPollInterval = 500 * time.Millisecond
+	// readyCheckTimeout determines how long to wait until giving up on waiting for
+	// BuildBuddy server to become ready. If this timeout is reached, the test case
+	// running the server will fail with a timeout error.
+	readyCheckTimeout = 30 * time.Second
+)
+
+type Server struct {
+	monitoringPort        int
+	healthCheckServerType string
+	mu                    sync.Mutex
+	exited                bool
+	// err is the error returned by `cmd.Wait()`.
+	err error
+}
+
+func runfile(t *testing.T, path string) string {
+	resolvedPath, err := bazelgo.Runfile(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return resolvedPath
+}
+
+type Opts struct {
+	BinaryPath            string
+	Args                  []string
+	HTTPPort              int
+	HealthCheckServerType string
+}
+
+func Run(t *testing.T, opts *Opts) *Server {
+	server := &Server{
+		monitoringPort:        opts.HTTPPort,
+		healthCheckServerType: opts.HealthCheckServerType,
+	}
+
+	cmd := exec.Command(runfile(t, opts.BinaryPath), opts.Args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Start(); err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		cmd.Process.Kill() // ignore errors
```

Comment:
nit: should we change this to `cmd.Process.Signal(syscall.SIGTERM)` to allow shutdown logic to run?

Reply:
> nit: should we change this to `cmd.Process.Signal(syscall.SIGTERM)` to allow shutdown logic to run?

I don't see the value. Since this is after the test is done, it doesn't matter if the binary shuts down gracefully or not.